pkg/crosscluster/logical/logical_replication_job.go

```diff
@@ -887,14 +890,50 @@ func (r *logicalReplicationResumer) ingestWithRetries(
 	ctx context.Context, execCtx sql.JobExecContext,
 ) error {
 	ingestionJob := r.job
+	details := ingestionJob.Details().(jobspb.LogicalReplicationDetails)
 	ro := getRetryPolicy(execCtx.ExecCfg().StreamingTestingKnobs)
 	var err error
 	var lastReplicatedTime hlc.Timestamp
+	resolvedDest, err := resolveDest(ctx, execCtx.ExecCfg(), details.SourceClusterConnUri)
+	if err != nil {
+		log.Warningf(ctx, "failed to resolve destination details for change monitoring: %v", err)
+	}
+
 	for retrier := retry.Start(ro); retrier.Next(); {
-		err = r.ingest(ctx, execCtx)
+		confPoller := make(chan struct{})
+		g := ctxgroup.WithContext(ctx)
+		g.GoCtx(func(ctx context.Context) error {
+			defer close(confPoller)
+			return r.ingest(ctx, execCtx)
+		})
+		g.GoCtx(func(ctx context.Context) error {
+			t := time.NewTicker(15 * time.Second)
+			defer t.Stop()
+			for {
+				select {
+				case <-ctx.Done():
+					return ctx.Err()
+				case <-confPoller:
+					return nil
+				case <-t.C:
+					newDest, err := reloadDest(ctx, ingestionJob.ID(), execCtx.ExecCfg())
+					if err != nil {
+						log.Warningf(ctx, "failed to check for updated configuration: %v", err)
+					} else if newDest != resolvedDest {
+						resolvedDest = newDest
+						return replanErr
+					}
+				}
+			}
+		})
+		err = g.Wait()
 		if err == nil {
 			break
 		}
+		if errors.Is(err, replanErr) {
+			log.Infof(ctx, "retry ingest due to updated configuration")
```

Comment:
I wonder if we should `retrier.Reset()` here so that these don't count against the retry backoff?